mobile/lib/screens/dashboard/dashboard_view.dart

```diff
@@ -221,59 +223,81 @@ class _DashboardViewState extends State<DashboardView>
                 children: [
                   BlocBuilder<NearbyLocationBloc, NearbyLocationState>(
                     builder: (context, state) {
+                      CurrentLocation? currentLocation = state.currentLocation;
                       switch (state.blocStatus) {
                         case NearbyLocationStatus.searchComplete:
                           break;
                         case NearbyLocationStatus.searching:
-                          if (state.locationAirQuality == null) {
+                          if (currentLocation == null) {
                             return const SearchingAirQuality();
                           }
                           break;
-                        case NearbyLocationStatus.locationDenied:
-                          return const Padding(
-                            padding: EdgeInsets.only(top: 16),
-                            child: LocationDeniedButton(),
-                          );
                         case NearbyLocationStatus.locationDisabled:
                           return const Padding(
                             padding: EdgeInsets.only(top: 16),
-                            child: NoLocationAirQualityMessage(
-                              "Turn on location to get air quality near you.",
-                              dismiss: false,
-                            ),
+                            child: DashboardLocationButton(),
                           );
                       }
 
-                      final AirQualityReading? nearbyAirQuality =
-                          state.locationAirQuality;
-                      if (nearbyAirQuality == null) {
-                        _nearbyLocationExists = false;
-
+                      if (currentLocation == null) {
                         return state.showErrorMessage
                             ? const Padding(
                                 padding: EdgeInsets.only(top: 16),
                                 child: NoLocationAirQualityMessage(
-                                  "We’re unable to get your location’s air quality. Explore locations below as we expand our network.",
+                                  "We’re unable to get your current location. Explore locations below in the meantime.",
```

Comment:
these location recommendations in cases where we fail to get the nearest location, how are they generated?
Do we also consider distance? @NoahNsimbe

Reply:
No we don't consider distance. We show at most two location in all African cities.

Reply:
Oh, thanks @NoahNsimbe  

Could we consider "recommending" nearer locations? Say within the same country to say the least.
Nothing urgent but perhaps consider it in the next PRs?

**Why am I suggesting this?**
By the time someone goes on to search for a specific place, it is only fair that we recommend nearest places in case we do not find an exact match for their search.

Reply:
Thanks @Baalmart. We can attempt to add it to PR. @Baalmart